### ISSUE TITLE
Comments inside a cpp implementation are duplicated.

### ISSUE THREAD

**Issue by abn6** [doxygen/doxygen#11280]
# Description
Comments inside the implementation of a function in a `.cpp`  are duplicated. This only happens if:
- There are two files: one for implementation and one for declaration.
- The function is inside a namespace.
- The decalration is loaded before the implementation.

# Screenshots
- Without errors (loading `implementation.cpp` before `declaration.cpp`):
![image](https://github.com/user-attachments/assets/ab7e6c2f-4517-46b9-9d11-e2f4517bc20a)

- With errors (loading `declaration.cpp` before `implementation.cpp`):
![image](https://github.com/user-attachments/assets/2075b9de-460f-4911-a921-a4e9a32cdba2)

# To Reproduce
Self contained example: [single_func_test.zip](https://github.com/user-attachments/files/18164857/single_func_test.zip)
Steps to reproduce:
- run `doxygen`
- open html result (or run `start "" "html\index.html"`) - the issue should occur.
- Edit `Doxyfile`: 
  change 
  ```
  INPUT                  = declaration.cpp implementation.cpp
  ```
  with
  ```
  INPUT                  = implementation.cpp declaration.cpp
  ```
- run `doxygen`
- open html result (or run `start "" "html\index.html"`) - the issue should not occur.

# Expected behavior
The result should be the same regardles of the order in which doxygen loads the files.

# Version
- [Doxygen 1.12](https://github.com/doxygen/doxygen/releases/tag/Release_1_12_0) (commit c73f5d30f9e8b1df5ba15a1d064ff2067cbb8267)

# Additional context
When loading entire folders, there is no way of controlling which files are loaded first, so I cannot work around the issue.

**Comment by albert-github** [comment]
A strange aspect to me is that the declarations are in a `.cpp` and not in a `.h` file that is included with the implementation / other files where it is needed by means of a `#include`.

Maybe you should also have a look at the settings:
- `REPEAT_BRIEF`
- `HIDE_IN_BODY_DOCS`

**Comment by doxygen** [comment]
@abn6 Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by abn6** [comment]
Indeed, the problem is fixed with Doxygen 1.13.

However, the behaviour pointed out by @albert-github still occurs https://github.com/doxygen/doxygen/issues/11280#issuecomment-2558434727.
